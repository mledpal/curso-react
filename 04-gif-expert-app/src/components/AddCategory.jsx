import { useState } from 'react';

export const AddCategory = ( { onAddCategory } ) => { 

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( { target } ) => {        
        setInputValue( target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();

        const newCategory = inputValue.trim();

        if( newCategory.length <= 1 ) return;

        onAddCategory( newCategory );
        setInputValue('');
    };

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text" 
                placeholder ="Buscar GIFs"
                value = { inputValue }
                onChange = { onInputChange }
            />
        </form>
    )    
}