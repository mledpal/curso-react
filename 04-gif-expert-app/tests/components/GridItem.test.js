import { render, screen } from "@testing-library/react";
import { GridItem } from "../../src/components/GridItem";

describe('Probando GridItem', () => {

    const title = 'Esto es un titulo';
    const url = 'http://miurl.com/';

    test('Debe de mostrar el componente correctamente', () => {
        const { container } = render(<GridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el ALT y el SRC adecuado', () => {

        render(<GridItem url={url} title={title} />);
        const { src, alt } = screen.getByRole('img');

        // screen.debug();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el parrafo como un texto', () => {
        render(<GridItem url={url} title={title} />);
        expect(screen.getByText(title)).toBeTruthy();

    });

})