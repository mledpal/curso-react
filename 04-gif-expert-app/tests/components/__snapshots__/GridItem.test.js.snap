// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GridItem Debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Esto es un titulo"
      src="http://miurl.com/"
    />
    <p>
      Esto es un titulo
       
    </p>
  </div>
</div>
`;
